### DISCUSSION TITLE
Decoding frames from the Monitor-2 satellite

### DISCUSSION THREAD

**Discussion by Foxiks** [daniestevez/gr-satellites#533] [Ideas]
Hello!
I tried to demodulate a dump with data from the Monitor-2 satellite at a frequency of 435.815 MHz. And it seems to have worked out for me.

![image](https://github.com/daniestevez/gr-satellites/assets/61653684/88aef0ff-46d4-46f4-a3f8-918d37e8c2d9)

I made a simple flowgraph that decodes a signal into a raw bitstream.

After receiving the bits of information, I used a deframer with the sync word ```0xb1d2769bacf51f2087``` and this allowed me to get transport packets with a length of ```1136 bits```.

Could you add support for this satellite to gr-satellites?

**Comment by daniestevez** [comment]
Hi @Foxiks, can you give more details or link to documentation about this satellite? I would need information about scrambling, FEC, CRC, etc. Do some of the other Russian satellites use the same modulation and coding (see #530, #531, #532)?

**Reply by Foxiks** [reply]
Unfortunately I do not have documentation for this type of signal.

**Comment by janvgils** [comment]
The FM/PM part seems to be Doka, so apart from AFSK AX.25 the satellite is also dumping data using the [Doka](https://pe0sat.vgnet.nl/download/DOKA/) scheme.

If you follow this [link](https://pe0sat.vgnet.nl/download/DOKA/), you can find some information that I found in the past.

**Comment by janvgils** [comment]
An example observation with both AFSK and FSK https://network.satnogs.org/observations/8874840/

**Comment by K4KDR** [comment]
The FSK at the beginning of that SatNogs observation appears to be 9600-something... if you just use the 'FSK Demodulator' block and stream the resulting bytes into a .bin file, there are a lot of patterns that repeat throughout.

![Screenshot from 2024-01-15 20-19-49](https://github.com/daniestevez/gr-satellites/assets/19886368/e2fba414-4084-42b6-afa8-2bae70d712c0)

... this was obtained using:

![Screenshot from 2024-01-15 20-20-25](https://github.com/daniestevez/gr-satellites/assets/19886368/e5c16726-2b17-4f90-8365-fabd084e77af)


... but I could not get a match from any of the deframers at my disposal.

-Scott
